redeem/plugins/UserInterruptPlugin.py

```diff
@@ -0,0 +1,147 @@
+"""
+Plugin for using one of the end stop inputs as an emergecy stop button.
+
+Author: Chad Hinton heavily leveraging the work of Elias Bakken from StartButtonPlugin.py
+email: chadwhinton(at)me(dot)com
+License: GNU GPL v3: http://www.gnu.org/copyleft/gpl.html
+
+ Redeem is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Redeem is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Redeem.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from AbstractPlugin import AbstractPlugin
+import logging
+import os
+try:
+    from Printer import Printer
+    from Key_pin import Key_pin
+except ImportError:
+    from redeem.Printer import Printer
+    from redeem.Key_pin import Key_pin
+
+__PLUGIN_NAME__ = 'UserInterrupt'
+
+
+class UserInterruptPlugin(AbstractPlugin):
+
+    @staticmethod
+    def get_description():
+        return "Plugin to repurpose an end stop input as an user interrupt button."
+
+    def __init__(self, printer):
+        super(UserInterruptPlugin, self).__init__(printer)
+
+        logging.debug('Activating '+__PLUGIN_NAME__+' plugin...')
+
+        if not self.printer.config.has_section(type(self).__name__):
+            logging.error(__name__+": Missing section in config file: "+type(self).__name__)
+        
+        if not self.printer.config.has_option(type(self).__name__, 'end_stop_input'):
+            logging.error(__name__+": Missing option: 'end_stop_input' ")
+            return
```

Comment:
Rather than "return", it would be better to perform all of the checks and then exit. Otherwise, the user is stuck in an incremental "fix one thing and retry" mode.  Better to help them fix all at once.